Automattic/wp-calypso
# Marketplace Thank You: Query data for Themes

Resolves #74878 

## Proposed Changes

* Query data read by selectors. This includes:
  * Products list
  * Purchases
  * Active Theme
* Instead of hiding an invalid theme. Display the theme with the action buttons disabled. This minimizes the layout shifts when loading the products and purchases for valid themes.

## Testing Instructions



* Navigate to the Marketplace Thank You page using this URL: `/marketplace/thank-you/:siteId?themes=:themeId`
* Using a `:siteId` that is Business or above and a `:themeId` that is free or you have purchased for that `:siteId`
* Activate the Theme
* Open the Developer Tools > Application and delete the IndexedDB
* Refresh the page
* Make sure you can see the buttons `Customize this design` meaning the Theme is considered `Active`
* Change the URL and add a theme that you have not purchased, e.g. `/marketplace/thank-you/:siteId?themes=lincoln`
* Make sure the CTA buttons for the Theme are disabled

### Regression testing
- To test the small update in `QueryProductsList` component. 
- Navigate to Themes and do some queries
- Make sure you do not see unexpected errors in the console

## Pre-merge Checklist
